packages/host/app/components/matrix/room.gts

```diff
@@ -206,6 +236,24 @@ export default class Room extends Component<Signature> {
 
   private doSendMessage = enqueueTask(
     async (message: string | undefined, cards?: CardDef[]) => {
+      let roomModule = await this.matrixService.getRoomModule();
+      let roomMember = new roomModule.RoomMemberField({
+        id: this.matrixService.userId,
+        userId: this.matrixService.userId,
+        roomId: this.args.roomId,
+      });
+      let externalId = String(this.room?.messages.length ?? 0);
+      let lastMessageSent = new roomModule.MessageField({
+        author: roomMember,
+        message,
+        formattedMessage: message ? sanitizeHtml(marked(message)) : '',
+        created: new Date().getTime(),
+        externalId: externalId,
+        transactionId: null,
+        attachedCardIds: cards?.map((c) => c.id) || [],
+      });
+      this.lastMessagesSent.set(this.args.roomId, lastMessageSent);
```

Comment:
I'm having some confusion figuring out how this works and why `lastMessagesSent` is needed here... 

I am wondering about an alternative approach, something like:

1. When the user sends a message, create a `userPendingMessage`, and set it to the conversation component. Give it a random client generated id. This message is not really a room's message, it is just a placeholder so that you can display the pending state. Include that generated id in the matrix payload to create a message. 

2. Once you detect there exists a room's message with the client generated id that you assigned to the pending message, remove the pending message.